sec_certs/dataset/fips.py

```diff
@@ -1,572 +1,339 @@
+        "fips_modules_historical.html": constants.FIPS_HISTORICAL_MODULES_URL,
+        "fips_modules_revoked.html": constants.FIPS_REVOKED_MODULES_URL,
+    }
 
     @property
-    def _policies_dir(self) -> Path:
-        return self.root_dir / "security_policies"
+    def policies_dir(self) -> Path:
+        return self.certs_dir / "policies"
 
     @property
-    def _algs_dir(self) -> Path:
-        return self.web_dir / "algorithms"
+    def policies_pdf_dir(self) -> Path:
+        return self.policies_dir / "pdf"
 
-    def _get_certs_from_name(self, module_name: str) -> List[FIPSCertificate]:
-        """
-        Returns list of certificates that match given name.
+    @property
+    def policies_txt_dir(self) -> Path:
+        return self.policies_dir / "txt"
 
-        :param str module_name: name to search for
-        :return List[FIPSCertificate]: List of certificates with web_data.module_name == module_name
-        """
-        return [crt for crt in self if crt.web_data.module_name == module_name]
+    @property
+    def module_dir(self) -> Path:
+        return self.certs_dir / "modules"
 
-    @serialize
-    def pdf_scan(self, redo: bool = False) -> None:
-        """
-        pdf_scan()
-        Extracts data from pdf files
+    @property
+    def algorithms_dir(self) -> Path:
+        return self.auxillary_datasets_dir / "algorithms"
+
+    @property
+    def algorithm_dataset_path(self) -> Path:
+        return self.algorithms_dir / "algorithms.json"
+
+    def __getitem__(self, item: str) -> FIPSCertificate:
+        if len(item) < 5:
+            return super().__getitem__(fips_dgst(item))
+        else:
+            return super().__getitem__(item)
 
-        :param bool redo: Whether to try again with failed files, defaults to False
+    def _extract_data_from_html_modules(self) -> None:
         """
-        logger.info("Entering PDF scan.")
+        Extracts data from html module file
+        :param bool fresh: if all certs should be processed, or only the failed ones. Defaults to True
+        """
+        logger.info("Extracting data from html modules.")
+        certs_to_process = [x for x in self if x.state.module_is_ok_to_analyze()]
+        processed_certs = cert_processing.process_parallel(
+            FIPSCertificate.parse_html_module,
+            certs_to_process,
+            config.n_threads,
+            use_threading=False,
+            progress_bar_desc="Extracting data from html modules",
+        )
+        self.update_with_certs(processed_certs)
 
-        keywords = cert_processing.process_parallel(
-            FIPSCertificate.find_keywords,
-            [cert for cert in self.certs.values() if not cert.pdf_data.keywords or redo],
+    @serialize
+    def _extract_data(self) -> None:
+        logger.info("Extracting various data from certification artifacts.")
+        for cert in self:
+            cert.state.policy_extract_ok = True
```

Comment:
Well the status is set to ok and then updated to False if it fails. Certs cannot change it, because they don't know whether the processing was invoked only on the failed certificates (in that cases, certs with `something_extract_ok = True` are discarded).